source/AI.cpp

```diff
@@ -1686,7 +1688,7 @@ bool AI::DoHarvesting(Ship &ship, Command &command)
 			
 			double degreesToTurn = TO_DEG * acos(min(1., max(-1., p.Unit().Dot(ship.Facing().Unit()))));
 			time += degreesToTurn / ship.TurnRate();
-			if(time < bestTime)
+			if(time < bestTime && ship.Cargo().Free() >= it->UnitSize())
```

Comment:
Since checking if you have space free is much less computationally intensive than calculating the approximate intercept time, this check should probably go at the very top of the for loop, i.e.:

```C++
for(const shared_ptr<Flotsam> &it : flotsam)
{
    if(ship.Cargo().Free() < it->UnitSize())
        continue;
    ...
```